packages/vite-plugin-svelte/src/utils/options.ts

```diff
@@ -324,9 +325,18 @@ export async function buildExtraViteConfig(
 	options: PreResolvedOptions,
 	config: UserConfig
 ): Promise<Partial<UserConfig>> {
+	// make sure we only readd vite default mainFields when no other plugin has changed the config already
+	// see https://github.com/sveltejs/vite-plugin-svelte/issues/581
+	if (!config.resolve) {
+		config.resolve = {};
+	}
+	config.resolve.mainFields = [
+		...SVELTE_RESOLVE_MAIN_FIELDS,
+		...(config.resolve.mainFields ?? VITE_RESOLVE_MAIN_FIELDS)
+	];
+
 	const extraViteConfig: Partial<UserConfig> = {
 		resolve: {
-			mainFields: [...SVELTE_RESOLVE_MAIN_FIELDS],
```

Comment:
Me being picky 😬 But I think it'll be a bit cleaner if we put this inside `extraViteConfig` like before with:

```js
mainFields: [
  ...SVELTE_RESOLVE_MAIN_FIELDS,
	...(config.resolve?.mainFields ?? VITE_RESOLVE_MAIN_FIELDS)
]
```

Reply:
Nvm looks like we need to mutate it, since returning an array would have Vite merging the arrays.